### PR TITLE
Implemented conditional testing in pipeline

### PR DESCRIPTION
**Pull request by Ben-EJ**

### CODE REVIEW

File: .github/workflows/pr-build.yml [hunk 1]
```diff
@@ -7,7 +7,30 @@ on:
       - 'main'
 
 jobs:
-  run-pr-checks:
+  changes:
```
Comment (amehta-scottlogic) [inline .github/workflows/pr-build.yml:10]: Minor: can we name this detect-change-scope (or something better if you can think of it). Just so it's obvious what we're doing

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -7,7 +7,30 @@ on:
...
+      with:
+        filters: |
+          etl:
+            - 'air-quality-backend/src/etl/**'
+          api:
+            - 'air-quality-backend/src/api/**'
+          etl_tests:
+            - 'air-quality-backend/tests/etl_tests/**'
+          api_tests:
+            - 'air-quality-backend/tests/api_tests/**'
```
Comment (amehta-scottlogic) [inline .github/workflows/pr-build.yml:31]: This is nice.

Should we combine etl + etl_tests together (and same for api). E.g.

```
          etl:
            - 'air-quality-backend/src/etl/**'
            - 'air-quality-backend/tests/etl_tests/**'
          api:
            - 'air-quality-backend/src/api/**'
            - 'air-quality-backend/tests/api_tests/**'
```

This way we don't need as big an if statement below - and I can't see any reason to keep them separate?

File: .github/workflows/pr-build.yml [hunk 3]
```diff
@@ -20,18 +43,66 @@ jobs:
...
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          channels: conda-forge
+          environment-file: air-quality-backend/conda/api_environment.yml
+          auto-activate-base: false
+          activate-environment: api-dev
+         
+      - name: Run api unit tests
+        run: python -m pytest --cov=src --cov=scripts --cov-report=xml:air-quality-backend/tests/coverage/api_test_report.xml air-quality-backend/tests/api_tests
```
Comment (amehta-scottlogic) [inline .github/workflows/pr-build.yml:101]: Can you uncomment your unit test now and see if it passes?

File: .github/workflows/pr-build.yml [hunk 4]
```diff
@@ -20,18 +43,66 @@ jobs:
...
+
       - name: Code Coverage Report
         uses: orgoro/coverage@v3.1
         with:
-            coverageFile: air-quality-backend/tests/coverage/test_report.xml
+            coverageFile: air-quality-backend/tests/coverage/api_test_report.xml
             token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (amehta-scottlogic) [inline .github/workflows/pr-build.yml:107]: I wonder what happens if we have two coverage reports on the PR? Maybe it shows both?

Comment (Ben-EJ) [reply .github/workflows/pr-build.yml:103]: Hmnnn not sure

**Pull request closed without merging**